frontend/app/players/page.js

```diff
@@ -0,0 +1,198 @@
+            <button className='btn btn-outline-secondary' type='button' onClick={() => AddPlayer(playerName)}>Add player</button>
+          </div>
+          <div className='form-text mb-2'>Add a new player?</div>
+        </div>
+
+        {/* Table */}
+        <div>
+          <table className='table'>
+            <thead>
+              <tr>
+                <th scope='col'>Name</th>
+                <th scope='col'>Win Rate</th>
+              </tr>
+            </thead>
+            <tbody>
+              <PlayerData players={players} playerStats={playerStats} />
+            </tbody>
+          </table>
+        </div>
+
+        {/* Modal */}
+        <PlayerModal playerName={playerName} setPlayerName={setPlayerName} playerRequested={playerRequested} setPlayerRequested={setPlayerRequested} />
+
+      </div>
+    );
+  } else {
+    console.log(playersError, matchesError);
+  }
+}
+
+function calculatePlayerStats(players, matches) {
+  const playerStats = {};
+
+  players.forEach(({ pname }) => {
+    if (!playerStats[pname]) {
+      playerStats[pname] = { wins: 0, matches: 0, winrate: 0, titles: [] };
+    }
+  });
+
+  matches.forEach(({ pname, team, ascore, bscore }) => {
+    const player = playerStats[pname];
+    player.matches += 1;
+    if ((team === 'a' && ascore > bscore) || (team === 'b' && bscore > ascore)) {
+      player.wins += 1;
+    }
+  });
+
+  for (const p in playerStats) {
+    const player = playerStats[p];
+    player.winrate = player.wins / player.matches;
+    if (isNaN(player.winrate)) player.winrate = 'N/A';
+    if (player.matches < 5) player.titles.push('Sprout');
+    if (player.matches > 30) player.titles.push('Veteran');
+    if (player.matches > 100) player.titles.push('Legend');
+    if (player.winrate >= 0.75) player.titles.push('Farmer');
+  }
+
+  return playerStats;
+}
+
+
+function PlayerData({ players, playerStats }) {
+  const titleColors = {
+    Sprout: 'text-bg-success',
+    Veteran: 'text-bg-secondary',
+    Legend: 'text-bg-primary',
+    Farmer: 'text-bg-danger'
+  };
+
+  const titleTooltips = {
+    Sprout: 'New (matches < 5)',
+    Veteran: 'Seasoned (matches > 30)',
+    Legend: 'Master (matches > 100)',
+    Farmer: 'Bully (win rate > 75%)'
+  };
+
+  return (
+    players.map(({ id, pname }) => (
+      <tr key={id}>
+        <td>
+          {pname}
+          {playerStats[pname]?.titles.map((title, index) => (
+            <OverlayTrigger
+              key={index}
+              overlay={<Tooltip id={`tooltip-${title}`} style={{ fontSize: '12px' }}>{titleTooltips[title]}</Tooltip>}
+            >
+              <span className={`badge rounded-pill ${titleColors[title]} ms-2`}>{title}</span>
+            </OverlayTrigger>
+          ))}
+        </td>
+        <td>
+          {playerStats[pname].winrate.toLocaleString(undefined, { style: 'percent', minimumFractionDigits: 2 })}
+          <span style={{ fontSize: '12px' }}> {'('}{playerStats[pname]?.matches || 0}{')'}
```

Comment:
It's a little unclear what exactly this number means when viewed on the page itself (could be interpreted as total wins, total matches, etc.). Mayhaps you can change it to a fraction (wins/total)?

Reply:
done